sympy/polys/ring_series.py

```diff
@@ -1158,3 +1158,97 @@ def rs_compose_add(p1, p2):
     if dp:
         q = q*x**dp
     return q
+
+
+class RingSeriesBase(object):
```

Comment:
All args of a class inheriting from ``Basic`` or it's derivative should be instances of ``Basic``(core/tests/test_args.py). This can cause some problems. Add some tests. It can be decided after seeing what all you want to do.